stdlib/sys.mli

```diff
@@ -20,6 +20,13 @@
   an error.
 *)
 
+type backend_type = 
+  | Native
+  | Bytecode
+  | Other of string 
```

Comment:
The problem with an extensible variant type is that you would need to link against the module which declares it, which is not good.  For instance,  imagine an OCaml library that wants to use a different algorithm when compiled to Javascript (e.g. because it knows that exceptions are slow there).  It needs to be able to test the "backend type" without creating a link dependency to e.g. js_of_ocaml.